File: samples/Test/Classes/Control/ImageViewTest.cpp

#include "ImageViewTest.h"

ImageViewTest::ImageViewTest()
:m_iIndex(0)
{
    this->setTitle("CAImageView");
}

ImageViewTest::~ImageViewTest()
{

}

void ImageViewTest::viewDidLoad()
{
    CAView* view = CAView::createWithLayout(DLayout(DHorizontalLayout_L_R(50, 50), DVerticalLayout_T_B(50, 80)));
    view->setColor(CAColor4B::GRAY);
    this->getView()->addSubview(view);
    
    if (m_iIndex == 0)
    {
        CAImageView* image2 = CAImageView::createWithImage(CAImage::create("image/h1.png"));
        image2->setScaleType(CAImageView::ScaleType::FitImageXY); //default
        image2->setLayout(DLayoutFill);
        view->addSubview(image2);
    }
    else if (m_iIndex == 1)
    {
        CAImageView* image3 = CAImageView::createWithImage(CAImage::create("image/h1.png"));
        image3->setScaleType(CAImageView::ScaleType::FitImageCrop);
        image3->setLayout(DLayoutFill);
        view->addSubview(image3);
    }
    else if (m_iIndex == 2)
    {
        CAImageView* image4 = CAImageView::createWithImage(CAImage::create("image/h1.png"));
        image4->setScaleType(CAImageView::ScaleType::FitImageInside);
        image4->setLayout(DLayoutFill);
        view->addSubview(image4);
    }
    else
    {
        CAVector<CAImage* > img;
        for (int i=0; i<8; i++)
        {
            char temp[50];
            sprintf(temp, "animation/npc_382-%d.png",i+1);
            CAImage* im = CAImage::create(temp);
            img.pushBack(im);
        }
        CAImageView* image7 = CAImageView::createWithLayout(DLayout(DHorizontalLayout_W_C(96, 0.5f), DVerticalLayout_H_C(176, 0.5f)));
        image7->setAnimationImages(img);
        image7->setAnimationDuration(0.2);
        image7->startAnimating();
        view->addSubview(image7);
    }
    
    
    auto btn = CAButton::createWithLayout(DLayout(DHorizontalLayout_NW_C(0.3,0.5), DVerticalLayout_B_H(20 ,50)), CAButton::Type::RoundedRect);
    btn->setTitleForState(CAControl::State::Normal, "Switch Next");
    btn->addTarget([&] {
        
        ssize_t index = m_iIndex + 1;
        if (index > 3)
        {
            index = 0;
        }
        ImageViewTest* viewController = ImageViewTest::create();
        viewController->setIndex(index);
        RootWindow::getInstance()->getRootNavigationController()->replaceViewController(viewController, false);
        
    }, CAButton::Event::TouchUpInSide);
    this->getView()->addSubview(btn);

}

void ImageViewTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/ImageViewTest.js
/**
 * Created by zhanglei on 16/8/2.
 */
var ImageViewTest = ca.CAViewController.extend({

    ctor: function () {
        this._super();
        this._imageViewNum = 0;
    },

    viewDidLoad: function() {
        this.initImageView();
    },

    initImageView:function(){
        log("ImageViewText-->imageViewNum:"+this._imageViewNum);

        this.getView().removeAllSubviews();

        if (this._imageViewNum == 0)
        {
            var image1 = ca.CAImageView.createWithImage(ca.CAImage.create("image/h1.png"));
            image1.setLayout(ca.DLayout.fill());

            var view1 = ca.CAView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(50, 50), ca.DVerticalLayout_T_B(50, 50)));
            view1.addSubview(image1);
            this.getView().addSubview(view1);
        }
        else if (this._imageViewNum == 1)
        {
            var image2 = ca.CAImageView.createWithLayout(ca.DLayout.fill());

            image2.setImage(ca.CAImage.create("image/h1.png"));
            image2.setScaleType(ca.CAImageView.ScaleType.FitImageXY);
            var view2 = ca.CAView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(50, 50), ca.DVerticalLayout_T_B(50, 50)));
            view2.addSubview(image2);
            this.getView().addSubview(view2);
        }
        else if (this._imageViewNum == 2)
        {
            var image3 = ca.CAImageView.createWithLayout(ca.DLayout.fill());
            image3.setImage(ca.CAImage.create("image/h1.png"));
            image3.setScaleType(ca.CAImageView.ScaleType.FitImageCrop);

            var view3 = ca.CAView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(50, 50), ca.DVerticalLayout_T_B(50, 50)));
            view3.addSubview(image3);
            this.getView().addSubview(view3);
        }
        else if (this._imageViewNum == 3)
        {
            var image4 = ca.CAImageView.createWithLayout(ca.DLayout.fill());
            image4.setImage(ca.CAImage.create("image/h1.png"));
            image4.setScaleType(ca.CAImageView.ScaleType.FitImageInside);


            var view4 = ca.CAView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(50, 50), ca.DVerticalLayout_T_B(50, 50)));
            view4.addSubview(image4);
            this.getView().addSubview(view4);
        }
        else if (this._imageViewNum == 4)
        {
            var img = new Array(3);
            for (var i=0; i<8; i++)
            {

                var str = "animation/npc_382-"+ (i+1) +".png";
                img[i] = ca.CAImage.create(str);
            }
            var image7 = ca.CAImageView.createWithImage(ca.CAImage.create("animation/npc_382-1.png"));
            image7.setLayout(ca.DLayout.fill());
            image7.setAnimationImages(img);
            image7.setAnimationDuration(0.1);
            image7.startAnimating();

            var view7 = ca.CAView.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_L_R(100, 100), ca.DVerticalLayout_T_B(200, 200)));
            view7.addSubview(image7);
            this.getView().addSubview(view7);
        }

        var btn = ca.CAButton.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_NW_C(0.3,0.5) , ca.DVerticalLayout_B_H(20 ,50)),ca.CAButton.Type.RoundedRect);
        btn.setTitleForState(ca.CAControl.State.Normal, "Switch Next");
        btn.addTarget(this.btncallback.bind(this), ca.CAButton.Event.TouchUpInSide);
        this.getView().addSubview(btn);
    },

    btncallback: function ()
    {
        this._imageViewNum += 1;

        if(this._imageViewNum > 4)
        {
            this._imageViewNum = 0;
        }
        this.initImageView();
    }
});
